src/xrGame/game_cl_mp.cpp

```diff
@@ -1535,7 +1535,8 @@ void game_cl_mp::generate_file_name(string_path& file_name, LPCSTR file_suffix,
 #else
 void game_cl_mp::generate_file_name(string_path& file_name, LPCSTR file_suffix, time_t& date_time)
 {
-    xr_sprintf(file_name, "%s_%s", ctime(date_time), file_suffix);
+    std::time_t std_date_time = date_time;
```

Comment:
Why do you need this temporary variable?

Reply:
If I write "ctime(&(std::time_t)date_time)" - I'll get error:
src/xrGame/game_cl_mp.cpp:1539:42: error: cannot take the
      address of an rvalue of type 'std::time_t' (aka 'long')
    xr_sprintf(file_name, "%s_%s", ctime(&(std::time_t)std_date_time)...
So I based on this answers:
https://stackoverflow.com/questions/36576285/convert-long-int-to-const-time-t

Reply:
But the date_time variable is a reference to a type 'time_t' (I suggest you to mark is additionaly as const). So does it mean that 'std::time_t' and 'time_t' are different types here?

Reply:
Yes, different types here.
https://travis-ci.org/OpenXRay/xray-16/jobs/468604698#L3347 - gcc warning says "invalid conversion from ‘time_t {aka long int}’ to ‘const time_t* {aka const long int*}’ [-fpermissive]"

Reply:
The types are not so different, but the solution looks ugly. And ifdef's, of course... I reimplement the function in PR #301, please check if the warning disappear.